vault-core/pom.xml

```diff
@@ -182,6 +182,11 @@
             <artifactId>org.osgi.framework</artifactId>
             <scope>provided</scope>
         </dependency>
+        <dependency>
```

Comment:
Having this dependency (although not new) for only one usage seems not really good.
Let us rather try to get rid of commons collection4 dependency in vault core.

Reply:
Good point, and I consirered that myself. It seems only be used for avoiding string duplicates in memory. Maybe that's something not relevant anymore in today's VMs.